source/language/types.rst

```diff
@@ -279,25 +279,53 @@ their argument expression in parentheses:
 
    .. table:: Built-in mathematical functions in OpenQASM3.
 
-      +--------+
-      | arccos |
-      +--------+
-      | arcsin |
-      +--------+
-      | arctan |
-      +--------+
-      | cos    |
-      +--------+
-      | exp    |
-      +--------+
-      | ln     |
-      +--------+
-      | sin    |
-      +--------+
-      | sqrt   |
-      +--------+
-      | tan    |
-      +--------+
+      +----------+--------------------------------+---------------------+
+      | Function | [Input Range/Type] , [...]     | [Output Range/Type] |
+      +==========+================================+=====================+
+      | arccos   | [-1, 1]                        | [0, pi rad]         |
+      +----------+--------------------------------+---------------------+
+      | arcsin   | [-1, 1]                        | [-pi/2, pi/2 rad]   |
+      +----------+--------------------------------+---------------------+
+      | arctan   | [-∞, Real, ∞]                  | [-pi/2, pi/2 rad]   |
+      +----------+--------------------------------+---------------------+
+      | ceiling  | [float], [float]               | [float]             |
+      +----------+--------------------------------+---------------------+
+      | cos      | [float]                        | [float]             |
+      +----------+--------------------------------+---------------------+
+      | div      | ??                             | ??                  |
+      +----------+--------------------------------+---------------------+
+      | exp      | [float]                        | [float]             |
+      |          |                                |                     |
+      |          | [complex]                      | [complex]           |
+      +----------+--------------------------------+---------------------+
+      | floor    | [float], [float]               | [float]             |
+      +----------+--------------------------------+---------------------+
+      | log *(e)*| [float]                        | [float]             |
+      +----------+--------------------------------+---------------------+
+      |  mod     | [int], [int]                   | [int]               |
+      |          |                                |                     |
+      |          | [float], [int,float]           | [float]             |
+      +----------+--------------------------------+---------------------+
+      | popcount | [bits, unsigned]               | [positive int]      |
```

Comment:
if it outputs a positive int, perhaps it just ought to output a `uint`?

Reply:
Probably true

Reply:
Done